src/main/java/com/mcmoddev/orespawn/impl/os3/BuilderLogicImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.mcmoddev.orespawn.impl.os3;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+import com.mcmoddev.orespawn.OreSpawn;
+import com.mcmoddev.orespawn.api.os3.BuilderLogic;
+import com.mcmoddev.orespawn.api.os3.DimensionBuilder;
+
+public class BuilderLogicImpl implements BuilderLogic {
+	private final Map<Integer, DimensionBuilder> dimensions;
+	
+	public BuilderLogicImpl() {
+		this.dimensions = new HashMap<>();
+	}
+	
+	@Override
+	public DimensionBuilder newDimensionBuilder(String name) {
+		switch(name.toLowerCase(Locale.ROOT)) {
+		case "overworld":

```

Comment:
make "overworld", "the nether" and "the end" shared constants, also provide "the overworld", "nether" and "end" for consistency.